From 3d7c42a689f15d084b27e8aacf62441d6efdda07 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Sun, 22 May 2016 10:43:16 +0300
Subject: [PATCH 158/239] xdb: update scripts to use open browse window to
 select different files

More easy to select the files using browse windows

Change-Id: Iac5cd99a3ce6f5cc2c961cbbd1a2529c542b16c0
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/29848
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Haim Boot <hayim@marvell.com>
---
 scripts/debugger/atf.xdb              | 19 ++++++++++++++-----
 scripts/debugger/binary_extension.xdb |  9 +++++++--
 scripts/debugger/bootloader.xdb       |  9 +++++++--
 3 files changed, 28 insertions(+), 9 deletions(-)

diff --git a/scripts/debugger/atf.xdb b/scripts/debugger/atf.xdb
index 3580af8..2c8a4cd 100644
--- a/scripts/debugger/atf.xdb
+++ b/scripts/debugger/atf.xdb
@@ -1,13 +1,22 @@
 if (@strcmp(@ATF_FIB_BIN_PATH,"EMPTY")==0) then
-	MESSAGE "Update the path of the ATF FIB binary, and reload the script"
-	EXIT
+	set value @ATF_FIB_BIN_PATH = @queryinputfile("select FIB bin file", "all files (*.*)|*.*");
+	if @strlen(@bin_ext_elf_path) == 0 then
+		!user hit cancel button
+		print "user abort\n"
+		MESSAGE "Update the path of the ATF FIB binary, and reload the script"
+		EXIT
+	end
 end
 if (@strcmp(@ATF_BL1_ELF_PATH,"EMPTY")==0) then
-	MESSAGE "Update the path of the BL1 ELF, and reload the script"
-	EXIT
+	set value @ATF_BL1_ELF_PATH = @queryinputfile("select BL1 ELF file", "all files (*.*)|*.*|elf files (*.afx;*.elf)|*.afx;*.elf||");
+	if @strlen(@ATF_BL1_ELF_PATH) == 0 then
+		!user hit cancel button
+		print "user abort\n"
+		MESSAGE "Update the path of the BL1 ELF, and reload the script"
+		EXIT
+	end
 end
 
-
 ! Set value atf_run = 1 if the user run this script
 set val @atf_run = 1
 
diff --git a/scripts/debugger/binary_extension.xdb b/scripts/debugger/binary_extension.xdb
index 53653f6..b904ed1 100644
--- a/scripts/debugger/binary_extension.xdb
+++ b/scripts/debugger/binary_extension.xdb
@@ -1,6 +1,11 @@
 if (@strcmp(@BIN_EXT_ELF_PATH,"EMPTY")==0) then
-	MESSAGE "Update the path of the binary extension, and reload the script"
-	EXIT
+	SET VALUE @BIN_EXT_ELF_PATH = @queryinputfile("Select BLE elf file", "All Files (*.*)|*.*|ELF Files (*.afx;*.elf)|*.afx;*.elf||");
+	IF @strlen(@BIN_EXT_ELF_PATH) == 0 then
+		!User hit cancel button
+		print "USER ABORT\n"
+		MESSAGE "Update the path of the binary extension, and reload the script"
+		EXIT
+	END
 end
 
 ! Set value bin_ext_run = 1 if the user run this script
diff --git a/scripts/debugger/bootloader.xdb b/scripts/debugger/bootloader.xdb
index bfbd27f..fc82339 100644
--- a/scripts/debugger/bootloader.xdb
+++ b/scripts/debugger/bootloader.xdb
@@ -1,6 +1,11 @@
 if (@strcmp(@BL_ELF_PATH,"EMPTY")==0) then
-	MESSAGE "Update the path of the Bootloader binary, and reload the script"
-	EXIT
+	set value @BL_ELF_PATH  = @queryinputfile("select BL ELF file", "all files (*.*)|*.*");
+	if @strlen(BL_ELF_PATH) == 0 then
+		!user hit cancel button
+		print "user abort\n"
+		MESSAGE "Update the path of the Bootloader binary, and reload the script"
+		EXIT
+	end
 end
 
 ! if the ATF script is not run, reset the board and stop at 0xffff0000
-- 
1.9.1

